src/tribler/core/components/component.py

```diff
@@ -2,48 +2,32 @@
 
 import logging
 import sys
+import time
 from asyncio import Event
 from typing import Optional, Set, TYPE_CHECKING, Type, Union
 
-if TYPE_CHECKING:
-    from tribler.core.components.session import Session, T
-
-
-class ComponentError(Exception):
-    pass
-
-
-class ComponentStartupException(ComponentError):
-    def __init__(self, component: Component, cause: Exception):
-        super().__init__(component.__class__.__name__)
-        self.component = component
-        self.__cause__ = cause
-
+from tribler.core.components.exceptions import ComponentStartupException, MissedDependency, NoneComponent
+from tribler.core.components.reporter.exception_handler import default_core_exception_handler
+from tribler.core.sentry_reporter.sentry_reporter import SentryReporter
 
-class MissedDependency(ComponentError):
-    def __init__(self, component: Component, dependency: Type[Component]):
-        msg = f'Missed dependency: {component.__class__.__name__} requires {dependency.__name__} to be active'
-        super().__init__(msg)
-        self.component = component
-        self.dependency = dependency
+STARTED = 'started'
+STARTING = 'starting...'
+FAILED = 'failed'
+WAITING_FOR = 'waiting for'
+SHUTTING_DOWN = 'shutting down'
+SHUT_DOWN = 'shut down'
+COMPONENTS_STATUS = 'components_status'
```

Comment:
I'm not sure we need to have these constants extracted:
  * We do not use them like enum items and do not perform `==` with them.
  * All of them, except `STARTING` and `FAILED`, are used in a single place only, and for `FAILED`, it may be better to have two different statuses to distinguish failings at the startup and shutdown phases.
  * They are not complete status strings, but just some fragments of string templates, and it is possible to have component statuses like `f'{STARTED} in {time.time() - start_time:.4f}s'` or `f'Stopping {self.name}: waiting for {dependants} to release it'`.

To me, extracting these strings looks like a complication that makes code harder to read. I prefer to inline them directly into the corresponding `_set_component_status(...)` calls.

Reply:
All strings are inlined.